sdkproject/Assets/Mapbox/Unity/Map/SnapMapToOrigin.cs

```diff
@@ -0,0 +1,27 @@
+using Mapbox.Unity.Map;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SnapMapToOrigin : MonoBehaviour
+{
+	private AbstractMap _map;
+	
+	void Start()
+	{
+		var _map = GetComponent<AbstractMap>();
+		if(_map == null)
+		{
+			Debug.Log("Snap Map to Origin script should be on the same object with AbstractMap");
+			return;
+		}
+
+		_map.OnHeightChanged += (s) =>
+		{
+			_map.Root.transform.position = new Vector3(
+				_map.Root.transform.position.x,
+				-s.CenterHeight * _map.Root.transform.localScale.x,
```

Comment:
@david-rhodes you're right, not sure why I did it like that there. I think I was thinking this more like  abstract map modifiers when I did this but even then, abstract map object doesn't have to be the root, so yes this is bad.